.travis.yml

```diff
@@ -1,30 +1,120 @@
+  - wget -c 'https://repo.mysql.com/mysql-apt-config_0.8.13-1_all.deb'
+  - echo "mysql-apt-config mysql-apt-config/select-server select $MYSQL_VERSION" | sudo debconf-set-selections
+  - echo "mysql-apt-config mysql-apt-config/select-tools select Enabled" | sudo debconf-set-selections
+  - echo "mysql-apt-config mysql-apt-config/select-product select Ok" | sudo debconf-set-selections
+  - echo "mysql-apt-config mysql-apt-config/tools-component string mysql-tools" | sudo debconf-set-selections
   - sudo dpkg -i mysql-apt-config_0.8.13-1_all.deb
+  # Percona tools
+  - wget https://repo.percona.com/apt/percona-release_latest.$(lsb_release -sc)_all.deb
   - sudo dpkg -i percona-release_latest.$(lsb_release -sc)_all.deb
   - sudo percona-release enable-only tools
-  - sudo apt-get update -q
-  - sudo apt-get install -q -y --allow-unauthenticated -o Dpkg::Options::=--force-confnew mysql-server
-  - sudo apt-get install percona-xtrabackup-80
+  - sudo apt update
+
+
+inc_install: &inc_install
+  # Build Deps
+  - sudo apt install -y libsnappy-dev
+  - sudo apt-get install -q -y --allow-unauthenticated -o Dpkg::Options::=--force-confnew mysql-server percona-xtrabackup-80
+  # Setup common python dependencies
+  - pip install -r requirements.txt
+  - pip install -r requirements.dev.txt
+  - pip install -e .
+
+inc_unittest: &inc_unittest
+  - make unittest
+
+jobs:
+  allow_failures:
+    - stage: lint
+      python: 3.6
+      env: [MYSQL_VERSION=mysql-5.7]
+      name: fmt
+      before_install: *inc_before_install
+      install: *inc_install
+      script:
+        - make fmt
+        - if [ $(git diff --name-only --diff-filter=ACMR | wc -l ) != 0 ]; then
+          echo "Reformatting failed! Please run make fmt on your commits and resubmit!" 1>&2 ;
+          git diff ;
+          exit 1 ;
+          fi
+
+    - stage: typecheck
+      python: 3.6
+      env: [MYSQL_VERSION=mysql-5.7]
+      name: fmt
+      before_install: *inc_before_install
+      install: *inc_install
+      script:
+        - make typecheck
+  include:
+    - stage: copyright
+      python: 3.6
+      env: [MYSQL_VERSION=mysql-5.7]
+      name: fmt
+      before_install: *inc_before_install
+      install: *inc_install
+      script:
+        - make copyright
+
+    # Run linters with our "primary" Python version.
+    - stage: lint
+      python: 3.6
+      env: [MYSQL_VERSION=mysql-5.7]
+      name: lint
+      before_install: *inc_before_install
+      install: *inc_install
+      script:
+        - make lint
+
+    # Matrix build unittests
+    - env: [MYSQL_VERSION=mysql-5.7]
+      python: '3.6'
+      stage: unittest
+      before_install: *inc_before_install
+      install: *inc_install
+      script: *inc_unittest
+
+    - env: [MYSQL_VERSION=mysql-8.0]
+      python: '3.6'
+      stage: unittest
+      before_install: *inc_before_install
+      install: *inc_install
+      script: *inc_unittest
+
+    - env: [MYSQL_VERSION=mysql-5.7]
```

Comment:
I would double check that this actually runs the tests with MySQL 5.7. It would be surprising if it did and those worked as older MySQL versions have never been tested and afair there were several required features that were not supposed to work with the older versions.

Reply:
Though looking at it most of the version 8 specific requirements might have been outside of MyHoard but I'd still be quite surprised if 5.7 just worked out of the box.

Reply:
But also I don't know if there's any point in adding this even if it did work. We're not committed to supporting MySQL 5.7 and the very first moment any new functionality wouldn't work on 5.7 we'd just drop it and if anyone reported an issue for 5.7 we wouldn't fix it. So I think it should be removed regardless.

Reply:
Not sure how I convinced myself it did, but set-selections is needed to prevent a prompt. Removed all the references to 5.7 in favor of 8.